keps/0023-enable-disable-features.md

```diff
@@ -0,0 +1,131 @@
+
+A `Feature` task will look like:
+
+```
+tasks:
+  - name: mirror-maker-configuration
+    kind: Feature
+    spec:
+      parameter: MIRROR_MAKER_ENABLED
+      resources:
+        - mirror-maker-cm.yaml
+  - name: mirror-maker
+    kind: Feature
+    spec:
+      parameter: MIRROR_MAKER_ENABLED
+      resources:
+        - mirror-maker.yaml
+```
+
+
+the parameter used in task would be defined in the parameters for the Operator, and must evaluate to a boolean value.
+```
+apiVersion: kudo.dev/v1beta1
+parameters:
+  - name: MIRROR_MAKER_ENABLED
+    default: "false"
+    trigger: mirror-maker-plan
+```
+
+And the plan triggered by the parameter will be the one using `Feature` tasks in their steps.
+
+```
+plans:
+  mirror-maker-plan:
+    strategy: serial
+    phases:
+      - name: deploy-mirror-maker
+        strategy: serial
+        steps:
+          - name: configuration
+            tasks:
+              - mirror-maker-configuration
+          - name: deploy
+            tasks:
+              - mirror-maker
+```
+
+In the tasks is where we can use the `Feature` kind. Which will apply the resources rendered by templates `mirror-maker.yaml` and `mirror-maker-cm.yaml` in case the value of parameter `MIRROR_MAKER_ENABLED` is true. 
+In case, the value is `false` it will delete the resources rendered by `mirror-maker.yaml` and `mirror-maker-cm.yaml`. 
+
+In case the parameter value is **true**:
+
+If there are objects to created the Task will be marked successful, once those are created.
+If there are no objects to be created the Task will be marked as successful. 
+
+In case the parameter value is **false**:
+
+If there are objects to be deleted the Task will be marked as successful, once those are marked for Deletion. 
+If there are no objects to be deleted the Task will be marked as successful. 
+
+
+### Limitations
+
+However the `Feature` task will have limitations like:
+
+- KUDO Operator must define the parameter which will be evaluated as a boolean.
+- The defined parameter must evaluate to a boolean value
+- Multiple parameters can trigger the same plan, but only one parameter will be evaluated
+
+For the last use case, e.g., we can have a parameter `MIRROR_MAKER_MEM` which triggers the `mirror-maker-plan` but
+the parameter which will decide in `Feature Task` if the kubernetes objects are created/updated or deleted will be `MIRROR_MAKER_ENABLED`
+
+## Alternatives
+
+There are two alternatives to this approach
+
+- Add feature enablement field in Instance and drop using parameters for enabling features in operators. 
+That would mean redesigning the instance updates architecture. As right now, this is the only way to updates the KUDO Instances. 
+But the impact of introducing the new field would be quite limiting as it won't provide much more than what we already have with parameters 
+
+- Reconcile the world. KUDO creates a state of an instance based on `Instance` and `OperatorVersion` objects. And on
```

Comment:
If that were always possible, why bother with serial phases/steps/tasks at all? If you could always define the outcome state and ignore individual steps and its order then you wouldn't need KUDO.

Reply:
state can include phases/steps/tasks. And reconciling a state doesn't mean to ignore the order to reach that state.